draft-barnes-acme.md

```diff
@@ -198,324 +212,362 @@ To revoke a certificate, the client simply sends a revocation request, signed wi
+ACME is structured as a REST application with a few types of resources:
 
-All other fields in an ACME message are defined by the type, as described below.  Unrecognized fields in ACME messages MUST be ignored.  Creators of ACME messages MUST NOT create messages with duplicate fields.  Parsers of ACME messages MAY be tolerant of duplicate fields, but the behavior of the protocol in this case is undefined.
+* Registration resources, representing information about an account key
+* Authorization resources, representing an account key's authorization to act for an identifier
+* Challenge resources, representing a challenge to prove control of an identifier
+* Certificate resources, representing issued certificates
+* A "new-registration" resource
+* A "new-authorization" resource
+* A "new-certificate" resource
 
+In general, the intent is for authorization and certificate resources to contain only public information, so that CAs may publish these resoruces to document what certificates have been issued and how they were authorized.  Non-public information, such as
+contact information, is stored in registration resources.
 
-## General Request/Response Lifecycle
+In order to accomplish ACME transactions, a client needs to have the server's new-registration, new-authorization, and new-ceritificate URIs; the remaining URIs are provided to the client as a result of requests to these URIs.  To simplify
+configuration, ACME uses the "next" link relation to indicate URI to contact for the next step in processing: From registration to authorization, and from authorization to certificate issuance.  In this way, a client need only be configured with the registration URI.
 
-Client-server interactions in ACME are logically request/response transactions, corresponding directly to HTTPS requests and responses.  The client sends a request message of a particular type, and the server sends response of a corresponding type.
+The "up" link relation is used with challenge resources to indicate the authorization resource to which a challenge belongs.  It is also used from certificate resources to indicate a resource from which the client may fetch a chain of CA certificates that could be used to validate the certificate in the original resource.
 
-All requests for a given ACME server are sent to the same HTTPS URI.  It is assumed that clients are configured with this URI out of band.  ACME requests MUST use the POST method, and since they carry JSON payloads, SHOULD set the Content-Type header field to "application/json".  ACME responses MUST be carried in HTTP responses with the status code 200.  ACME clients SHOULD follow HTTP redirects (301 or 302 responses), in case an ACME server is relocated.
+The following diagram illustrates the relations between resources on an ACME server.  The solid lines indicate link relations, and the dotted lines correspond to relationships expressed in other ways, e.g., the Location header in a 201 (Created) response.
```

Comment:
I think that Richard needs to clean this up.  A link relation is a link in the 5988 sense.  It doesn't matter where it appears, it can still be a link relation.  The header field vs. body distinction doesn't read on the subject.